File: 198-house-robber/house-robber.cpp
class Solution {
public:
    int rob(vector<int>& nums) {
        int n=nums.size();
        if(n==0) return 0;
        vector<int>dp(n);
        dp[0]=nums[0];
        if(n<=1) return nums[0];
        else if(n==2) return max(nums[0],nums[1]);
        dp[1]=max(nums[1],nums[0]);
        int amt=max(dp[1],dp[0]);
        for(int i=2;i<n;++i){
            dp[i]=max(dp[i-1],dp[i-2]+nums[i]);
        }
        return max(dp[n-1],dp[n-2]);
    }
};

File: 198-house-robber/house-robber.java
class Solution {
    public int rob(int[] nums) {
        int n=nums.length;
        int[] dp=new int[n];
        if(n==1) return nums[0];
        dp[0]=nums[0];
        dp[1]=Math.max(dp[0],nums[1]);
        // if(n==2) return Math.max(dp[0],dp[1]);
        for(int i=2;i<n;++i){                                                                  
            dp[i]=Math.max(dp[i-1],dp[i-2]+nums[i]);
        }
        return Math.max(dp[n-1],dp[n-2]);
    }
}